### PR TITLE
AVRO-1762: Velocity custom objects

### PR DESCRIPTION
**Pull request by redsk**
Refer to [Avro-1762](https://issues.apache.org/jira/browse/AVRO-1762?focusedCommentId=16846039&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16846039) for more information.  

Avro 1762 pull request rebased on master and with some tests improvements.

This PR is a rebased version of #71

### CODE REVIEW

**Comment by david-espinosa** [comment]
I think this feature is really needed.

**Comment by Fokko** [comment]
Can you rebase onto master?

**Comment by redsk** [comment]
> Can you rebase onto master?

Sure thing. Done!

File: lang/java/compiler/src/main/java/org/apache/avro/compiler/specific/SpecificCompiler.java [hunk 1]
```diff
@@ -136,6 +136,7 @@ void addLogicalTypeConversions(SpecificData specificData) {
   private boolean enableDecimalLogicalType = false;
   private final DateTimeLogicalTypeImplementation dateTimeLogicalTypeImplementation;
   private String suffix = ".java";
+  private Object[] additionalVelocityTools = new Object[0];
```
Comment (Fokko) [inline lang/java/compiler/src/main/java/org/apache/avro/compiler/specific/SpecificCompiler.java]: I would prefer it to make this a list.

Comment (redsk) [reply lang/java/compiler/src/main/java/org/apache/avro/compiler/specific/SpecificCompiler.java]: Sure. Changed to list.

**Comment by Fokko** [comment]
Thanks @redsk @embs

**Pull request merged**